Refactor analyzer templates and .env.example

changjian-wang opened the pull request:
…d' and update environment variable names for content understanding API

## Purpose
Refactor analyzer templates and .env.example

## Does this introduce a breaking change?
<!-- Mark one with an "x". -->
```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[x] Other... Please describe: Refactor analyzer templates and .env.example
```

Commit:
Refactor analyzer templates to replace 'scenario' with 'baseAnalyzerId' and update environment variable names for content understanding API

Commit:
Update environment variable names for Azure AI Content Understanding and adjust related configurations in notebooks

Commit:
Update environment variable names for Azure AI Content Understanding and adjust file paths in notebooks

chienyuanchang commented on notebooks/search_with_document_layout.ipynb:
```diff
@@ -44,8 +44,12 @@
+    "load_dotenv()\n",
+    "\n",
+    "AZURE_AI_CONTENT_UNDERSTANDING_ENDPOINT = os.getenv(\"AZURE_AI_CONTENT_UNDERSTANDING_ENDPOINT\")\n",
+    "AZURE_AI_CONTENT_UNDERSTANDING_API_VERSION = os.getenv(\"AZURE_AI_CONTENT_UNDERSTANDING_API_VERSION\") or \"2024-12-01-preview\"\n",
```
We will also need to update the default API version for all notebooks to `2025-05-01-preview`.

chienyuanchang commented on notebooks/search_with_visual_document.ipynb:
```diff
@@ -313,7 +313,7 @@
     "            # Collect formatted content from the figure\n",
     "            content_understanding_response = content_understanding_client.begin_analyze(ANALYZER_ID, figure_filepath)\n",
-    "            content_understanding_result = content_understanding_client.poll_result(content_understanding_response, timeout_seconds=1000)\n",
+    "            content_understanding_result = content_understanding_client.poll_result(content_understanding_response, timeout_seconds=1000 * 6000)\n",
```
Why do we need a timeout about 70 days?

changjian-wang replied:
Just for test. Sometimes, I need many hours to run this function. But recently need 20mins to completed this operation. I have changed it to 1800s, about half an hour.

chienyuanchang replied:
Then it should be `1800` and not `1000 * 1800`?

Commit:
Update video content understanding schema and refine scene description processing in notebook

Commit:
Fix video content extraction by updating the JSON path for segments in the processing function

Commit:
Update API version to 2025-05-01-preview across notebooks and README

- Changed default API version in README.md and all relevant notebooks to the latest preview version (2025-05-01-preview).
- Updated Python environment version in notebooks to 3.13.7.
- Adjusted timeout for content understanding result polling in search_with_visual_document.ipynb to 1800 seconds.

chienyuanchang commented on notebooks/search_with_video_webapp.ipynb:
```diff
@@ -167,8 +167,8 @@
-    "AZURE_AI_SERVICE_ENDPOINT = os.getenv(\"AZURE_AI_SERVICE_ENDPOINT\")\n",
-    "AZURE_AI_SERVICE_API_VERSION = os.getenv(\"AZURE_AI_SERVICE_API_VERSION\", \"2024-12-01-preview\")\n",
+    "AZURE_AI_CONTENT_UNDERSTANDING_ENDPOINT = os.getenv(\"AZURE_AI_CONTENT_UNDERSTANDING_ENDPOINT\")\n",
+    "AZURE_AI_CONTENT_UNDERSTANDING_API_VERSION = os.getenv(\"AZURE_AI_CONTENT_UNDERSTANDING_API_VERSION\", \"2025-05-01-preview\")\n",
```
I will update the changes for this notebook in another PR: https://github.com/Azure-Samples/azure-ai-search-with-content-understanding-python/pull/20. The current branch won't be able to run with `2025-05-01-preview`. Maybe we can set it as `AZURE_AI_CONTENT_UNDERSTANDING_API_VERSION = 2024-12-01-preview` for this PR update first.

changjian-wang replied:
Reverted to 2024-12-01-preview.

Commit:
Update Azure API version defaults to use environment variables for configuration

Commit:
Fix API version assignment and timeout duration in Content Understanding notebook

- Updated the Azure AI Content Understanding API version to a fixed value "2024-12-01-preview" in the search_with_video_webapp.ipynb.
- Adjusted the timeout duration for polling results in the search_with_visual_document.ipynb from 1000 * 1800 seconds to 1800 seconds for clarity and consistency.
